bootstrap/pkg/kfapp/ksonnet/ksonnet.go

```diff
@@ -80,19 +80,19 @@ func GetKfApp(kfdef *kfdefs.KfDef) kftypes.KfApp {
 	if goPathVar != "" {
 		_kfapp.Spec.Repo = re.ReplaceAllString(_kfapp.Spec.Repo, goPathVar+`$2`)
 	}
-	// build restConfig and apiConfig using $HOME/.kube/config if the file exist
-	_kfapp.restConfig = kftypes.GetConfig()
-	_kfapp.apiConfig = kftypes.GetKubeConfig()
 	return _kfapp
 }
 
 // Apply applies the ksonnet components to target k8s cluster.
 // Remind: Need to be thread-safe: this entry is share among kfctl and deploy app
 func (ksApp *ksApp) Apply(resources kftypes.ResourceEnum) error {
+	// build restConfig and apiConfig using $HOME/.kube/config if the file exist
+	ksApp.restConfig = kftypes.GetConfig()
+	ksApp.apiConfig = kftypes.GetKubeConfig()
```

Comment:
Can we have
```
ksApp.restConfig = kftypes.GetConfig()
ksApp.apiConfig = kftypes.GetKubeConfig()
```
back to part of ```func GetKfApp(kfdef *kfdefs.KfDef)```?

```Apply()``` need to be thread safe and cannot set credentials.
``` kftypes.GetConfig()``` will print info instead of throw error when cannot get credential during ```generate```